src/deque.py

```diff
@@ -0,0 +1,94 @@
+import sys
+
+
+class Node(object):
+    """Creates a Node."""
+    def __init__(self, value, next_up=None, behind=None):
+        self.value = value
+        self.next = next_up
+        self.behind = behind
+
+
+class Deque(object):
+    """Creates a Deque from value or list."""
+    def __init__(self, optional_values=[]):
+        self.head = None
+        self.tail = None
+        self.length = 0
+        self.is_empty = True
+        for x in optional_values:
+            self.add_first(x)
+
+    def add_last(self, value):
+        """Adds node to end of queue."""
+        self.tail = Node(value, self.tail)
+        if self.length == 0:
+            self.head = self.tail
+        if self.tail.next is not None:
+            self.tail.next.behind = self.tail
+        if self.is_empty is True:
+            self.is_empty = False
+        self.length += 1
+
+    def add_first(self, value):
+        """Adds node to head or start of queue."""
+        self.head = Node(value, None, self.head)
+        if self.length == 0:
+            self.tail = self.head
+        if self.head.behind is not None:
+            self.head.behind.next = self.head
+        if self.is_empty is True:
+            self.is_empty = False
+        self.length += 1
+
+    def delete_last(self):
+        """Removes node from tail of queue."""
+        if self.tail is None:
+            return "Queue is empty."
```

Comment:
Instead of returning a string, think about what python error should be returned.